clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/providers/TitusClusterProvider.groovy

```diff
@@ -195,6 +216,25 @@ class TitusClusterProvider implements ClusterProvider<TitusCluster>, ServerGroup
     serverGroup
   }
 
+  private Optional<Map<String, String>> getTaggedImageAttributes(String account, Job job) {
+    try {
+      // Infer registry from account configuration
+      String registry = TitusUtils.getRegistry(accountCredentialsProvider, account)
+      String key = DockerRegistryCacheKeys.getTaggedImageKey(registry, job.applicationName, job.version)
+      Set<CacheData> images = DockerRegistryProviderUtils
+        .getAllMatchingKeyPattern(cacheView, DockerRegistryCacheKeys.Namespace.TAGGED_IMAGE.getNs(), key)
+      List<Map<String, String>> allAttributes = images.collect {it.attributes }
+      if(allAttributes.size() >1) {
+        throw new TitusException("More than 1 tagged image found, Expected 1, but got ${allAttributes.size()}")
+      }
+      else return Optional.of(allAttributes.first())
+    }
+    catch (Exception e) {
+      log.error("Failed to fetch tagged image attributes ", e)
```

Comment:
is this the correct behavior on a failure here? what happens if we deploy with no attributes, we just see no build information?

Reply:
The idea here was if we can fetch the image attributes we populate `buildInfo.jenkins.number` , `buildInfo.jenkins.name` and `buildInfo.jenkins.host`. In case of an error or no attributes it would be `buildInfo.jenkins : {}` . There could be a fair amount of images without these attributes. Do you see any concern with this approach ?

Reply:
I'd at least like to see a differentiation between an ERROR (something unexpected happened looking things up) vs an expected case that there are no tags available (probably DEBUG)

Reply:
sounds good , i'll address this.